apps/examples/e2e/tests/test-focus.spec.ts

```diff
@@ -139,7 +139,7 @@ test.describe('Focus', () => {
 		await page.waitForTimeout(100) // takes 30ms or so to focus
 		await page.keyboard.press('d')
 
-		expect(await EditorA.$('.tlui-button[data-testid="tools.draw"][aria-checked="true"]')).not.toBe(
+		expect(await EditorA.$('.tlui-button[data-testid="tools.draw"][aria-checked="true"]')).toBe(
```

Comment:
I'm not sure if understand why the draw tool should become unselected here?

Reply:
yeah, it took me a little bit to figure out - it's a quirk of this test.
it's because the `autoFocus` prop is changing, so in `TldrawEditorWithReadyStore`→ `useLayoutEffect`, the changing prop causes the editor to be reset. you wouldn't _normally_ change the `autoFocus` prop to change focus, you'd want to programatically change the editor you already have.

Reply:
that said, maybe i should update `MultipleExample` to not change props but actually update the instance state to more properly address this.

Reply:
> yeah, it took me a little bit to figure out - it's a quirk of this test. it's because the `autoFocus` prop is changing, so in `TldrawEditorWithReadyStore`→ `useLayoutEffect`, the changing prop causes the editor to be reset. you wouldn't _normally_ change the `autoFocus` prop to change focus, you'd want to programatically change the editor you already have.

ouch 😂 At least for me it definitely sounds like a property you set once and should not change, so the change in behaviour is fine.

Yeah, updating the example sounds good.

Reply:
i reworked MultipleExample to not changethe prop

Reply:
i don't feel suuuuuper strongly here - happy to leave as is - but in general, I would look to the browser/react for how attributes mirroring built-in ones should behave.

In both of those, i believe setting this attribute on a pre-existing element does nothing, but it definitely doesn't clear the state of the items. in the browser, i think it's respected on page load, and in react it's respected when an element is first mounted.

Reply:
This is a good point, I reworked this little bit so that TldrawEditor only takes the initial auto focus value. Changing it shouldn't affect the editor.